src/qibolab/platform.py

```diff
@@ -350,15 +342,39 @@ def execute_pulse_sequence(self, sequence: PulseSequence, options: ExecutionPara
         result = {}
         for instrument in self.instruments:
             if isinstance(instrument, Controller):
-                new_result = instrument.play(self.qubits, sequence, options)
+                new_result = getattr(instrument, method)(self.qubits, sequences, options)
                 if isinstance(new_result, dict):
                     result.update(new_result)
                 elif new_result is not None:
                     # currently the result of QMSim is not a dict
                     result = new_result
         return result
 
-    def sweep(self, sequence, options, *sweepers):
+    def execute_pulse_sequence(self, sequences: PulseSequence, options: ExecutionParameters, **kwargs):
+        """
+        Args:
+            sequence (:class:`qibolab.pulses.PulseSequence`): Pulse sequences to execute.
+            options (:class:`qibolab.platforms.platform.ExecutionParameters`): Object holding the execution options.
+            **kwargs: May need them for something
+        Returns:
+            Readout results acquired by after execution.
+
+        """
+        return self._execute("play", sequences, options, **kwargs)
+
+    def execute_pulse_sequences(self, sequences: List[PulseSequence], options: ExecutionParameters, **kwargs):
+        """
+        Args:
+            sequence (List[:class:`qibolab.pulses.PulseSequence`]): Pulse sequences to execute.
+            options (:class:`qibolab.platforms.platform.ExecutionParameters`): Object holding the execution options.
+            **kwargs: May need them for something
+        Returns:
+            Readout results acquired by after execution.
+
+        """
+        return self._execute("play_sequences", sequences, options, **kwargs)
```

Comment:
Sorry for being late: I would switch from string methods to a boolean option, e.g. `multiple=False` or something like.

It's not urgent, we can do at any later stage. Just to remember to avoid passing strings around (for a binary option, boolean are usually the best, otherwise go for enums).

Reply:
This is just a relict from the old layout. For sure we should rewrite in a proper way. I would say that we can do this in a separate PR.